### PR TITLE
JS: Omit uninteresting nodes from path explanations

### PR DESCRIPTION
**Pull request by asger-semmle**
Hides nodes from the path explanation that aren't interesting. This includes:
- Phi nodes, refinements, variable capture nodes.
- Intermediate nodes in a `+`-based concatenation tree.

Evaluation underway.

### CODE REVIEW

File: javascript/ql/src/semmle/javascript/dataflow/Configuration.qll [hunk 1]
```diff
@@ -925,7 +953,11 @@ class SinkPathNode extends PathNode {
  */
 module PathGraph {
   /** Holds if `nd` is a node in the graph of data flow path explanations. */
-  query predicate nodes(PathNode nd) { any() }
+  query predicate nodes(PathNode nd) {
+    not nd.isHidden() or
+    nd instanceof SourcePathNode or
```
Comment (esben-semmle) [inline javascript/ql/src/semmle/javascript/dataflow/Configuration.qll]: Edge case: the `SourcePathNode` character is independent of the path. So you may display a hidden `SourcePathNode` in the middle of your path. I suppose we can live with that.

**Comment by asger-semmle** [comment]
[Evaluation](https://git.semmle.com/asger/dist-compare-reports/tree/domitian.ti.semmle.com_1551854764036) looks okay. After conferring with @xiemaisi I'll turn this into a 1.20 hotfix.

File: javascript/ql/src/semmle/javascript/dataflow/Configuration.qll (hunk 1 above)
Comment (asger-semmle) [reply javascript/ql/src/semmle/javascript/dataflow/Configuration.qll]: I'm fine with that.

**Comment by asger-semmle** [comment]
> Can we have some explicit tests for this feature?

Writing a test for this feels like copy/pasting the implementation into a test to check they are the same. The updated output from every path-problem is not enough?

**Pull request merged**